syft/grid.py

```diff
@@ -9,17 +12,17 @@ def __init__(self, *workers):
 
     def search(
         self, *query, verbose: bool = True, return_counter: bool = True
-    ) -> Tuple[Dict, Counter]:
+    ) -> Union[Tuple[Dict[Any, Any], CounterType], Dict[Any, Any]]:
         """Searches over a collection of workers, returning pointers to the results
         grouped by worker."""
 
-        tag_counter = Counter()
```

Comment:
Oh, I've just noticed you've used typing in the code itself, for pysyft codebase we use typing just for docstrings. Could you remove this before we merge it?

Reply:
So there will be times where a static type checker will need additional information about the types of variables beyond the annotations in function/method argument and return types (e.g. one of the issues in `syft/grid.py` was `"error: Need type annotation for 'tag_counter'"`). If we want to be able to run mypy without errors on the codebase, there are two paths: 

* Add an annotation in variable definitions where needed such that mypy can infer the types of variables it doesn't yet understand (the approach I took in this diff), or

* Add the same type information in a comment e.g. `tag_counter = Counter()  # type: CounterType[int]`

Either is a fine solution, though the latter might be a bit more approachable for a contributor who is familiar with Python but not yet type annotations. I didn't take this path as there is [a linting check](https://github.com/OpenMined/PySyft/blob/dev/.pre-commit-config.yaml#L21) disallowing type information in comments. We _could_ disable that if it's preferable - what do you think?

Reply:
Hey @redshiftzero
What changes need to be made on a repository wide scale to get mypy happy with the typing in the code itself (ie do we need to add these comments/annotations all over the place or will it just be isolated.)

I personally am biased towards not having them as comments as I think it looks a bit cleaner and is more in line with how we do it for the method definitions. That being said I would love to hear @iamtrask @LaRiffle @mari-linhares and @midokura-silvia 's opinions on this as well

Reply:
I agree with @robert-wagner. 

If these are isolated changes I'm pro supporting mypy by the cost of having some files with comments/static notations.

To be honest I don't have a strong preference for comments/static typing.